### PR TITLE
(692)Configure ESLint Member Ordering

### PR DESCRIPTION
**Pull request by Gweaton**
## Context

There's currently a bit of overhead involved in PRs where we have preferences for certain fields etc. to be sorted alphabetically - it would be great to pick these up automatically so we can catch these before PR review and save time.

## Changes in this PR

Configures ESLint Member ordering to order methods, fields etc in the following order:
```
"static-method",
"static-field",
"abstract-method",
"abstract-field",
"instance-field",
"constructor",
"public-instance-method",
"protected-instance-method",
"private-instance-method"
```
NB: This doesn't apply to methods in files that declared on classes, e.g. utils. I've been unable to configure this, but I think we'll have to live with it for a bit.

We've decided to turn this on as there can be a fair bit of overhead on PR review, and catching this kind of thing early can save us time in the long run.

This also fixes the alphabetical order of all groups in the offending classes.

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: .eslintrc.json
Comment (yndajas) [inline .eslintrc.json:61]: Which of the [supported options](https://typescript-eslint.io/rules/member-ordering/#all-supported-options) is this? (Are we missing getters and setters here?)

File: .eslintrc.json
Comment (yndajas) [inline .eslintrc.json:48]: I think this should go before `@typescript-eslint/no-use-before-define` - alphabetical rules

File: server/applicationVersion.ts
Comment (Gweaton) [inline server/applicationVersion.ts:13]: Is this change enforced by linting? If not, can it be?

File: integration_tests/e2e/app.cy.ts
Comment (Gweaton) [inline integration_tests/e2e/app.cy.ts:3]: I've found this commit (015c53af) a bit hard to review as it's changing so many files into classes in one go.

With the amount that's changing here, it's been a bit tricky to review carefully, but the approach looks good me and luckily we have a comprehensive test suite. With future changes like this that affect a big chunk of the codebase more structurally than reordering some keys etc, moving one class at a time would probably lighten the load as a reviewer.

No need to change anything this time, as rebasing will be a pain, but just something to bear in mind for next time!

**Review by Gweaton — COMMENTED** [review]
@yndajas thanks for taking this and finishing it off (particularly all those manual changes 😵).

I think this looks good to me - I can't approve as it's my own PR. Just had one question on the named exports stuff that'd be good to consider

**Review by Gweaton — COMMENTED** [review]
Looks good, thanks for sorting! You'll have to approve as it's technically my PR but you've got my approval.

**Review by yndajas — APPROVED** [review]
Sorted

**Pull request merged**
